bundles/io/org.eclipse.smarthome.io.transport.mqtt.test/src/test/java/org/eclipse/smarthome/io/transport/mqtt/test/MqttServiceTests.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.transport.mqtt.test;
+
+import static org.junit.Assert.*;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.Map;
+
+import javax.naming.ConfigurationException;
+
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection;
+import org.eclipse.smarthome.io.transport.mqtt.MqttBrokersObserver;
+import org.eclipse.smarthome.io.transport.mqtt.MqttService;
+import org.junit.Test;
+
+/**
+ * Tests the MqttService class
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class MqttServiceTests {
+    boolean brokerExisting = false;
+
+    // Tests addBrokersListener/removeBrokersListener
+    @Test
+    public void brokerConnectionListenerTests() {
+        MqttService service = new MqttService();
+        assertFalse(service.isBrokerObservers());
+        MqttBrokersObserver observer = new MqttBrokersObserver() {
+            @Override
+            public void brokerAdded(MqttBrokerConnection broker) {
+                brokerExisting = true;
+            }
+
+            @Override
+            public void brokerRemoved(MqttBrokerConnection broker) {
+                brokerExisting = false;
+            }
+        };
+        service.addBrokersListener(observer);
+        assertTrue(service.isBrokerObservers());
+
+        MqttBrokerConnection connection;
+        try {
+            connection = new MqttBrokerConnection("name", "tcp://123.123.123.123", false);
+        } catch (ConfigurationException c) {
+            fail("Couldn't create a MqttBrokerConnection object");
+            return;
+        }
+        assertTrue(service.addBrokerConnection(connection));
+        assertTrue(brokerExisting);
```

Comment:
Just like in the `MqttBrokerConnectionTests` this would be a little shorter with a mock `MqttBrokersObserver` and `org.mockito.Mockito.vrify`.